File: Java/lab4/src/main/java/org/example/factory/PcFactory.java
package org.example.factory;

import org.example.interfaces.Factory;
import org.example.interfaces.LaptopProduct;
import org.example.interfaces.PcProduct;
import org.example.interfaces.PhoneProduct;
import org.example.pc.GamingPc;
import org.example.pc.OficePc;
public class PcFactory implements Factory {
    @Override
    public LaptopProduct createLaptop(String typeOfLaptop) {
        return null;
    }
    public PcProduct createPc(String typeOfPc) {
        return switch (typeOfPc) {
            case "Gaming" -> new GamingPc();
            case "Ofice" -> new OficePc();
            default -> null;
        };
    }
    @Override
    public PhoneProduct createPhone(String typeOfPhone) {
        return null;
    }
}

File: Python/lab4/factory/PcFactory.py
from interfaces.PcProduct import PcProduct
from interfaces.Factory import Factory
from pc.GamingPc import GamingPc
from pc.OficePc import OficePc
from interfaces.PhoneProduct import PhoneProduct
from interfaces.LaptopProduct import LaptopProduct


class PcFactory(Factory):
    def createPc(self, type_of_pc) -> PcProduct:
        if type_of_pc == 'Gaming':
            return GamingPc()
        elif type_of_pc == "Ofice":
            return OficePc()
        else:
            return None

    def createLaptop(self, type_of_pc) -> LaptopProduct:
        return None

    def createPhone(self, type_of_phone) -> PhoneProduct:
        return None
